pkg/resource/layer_version/hooks.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package layer_version
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/lambda"
+)
+
+// customPreDelete deletes all the previous versions of a
+//LayerVersion except the latest version
+
+// This function is used as a sdk_delete hook, to delete
+// all the previous versions of a LayerVersion when delete API call is made
+func customPreDelete(
+	r *resource,
+	rm *resourceManager,
+	ctx context.Context,
+) error {
+	// Getting the list of all the versions of a LayerVersion
+	input := &svcsdk.ListLayerVersionsInput{
+		LayerName: r.ko.Spec.LayerName,
+	}
+	response, err := rm.sdkapi.ListLayerVersionsWithContext(ctx, input)
+	if err != nil {
+		return err
+	}
+
+	log := ackrtlog.FromContext(ctx)
+	log.Info("Deleting previous versions of LayerVersion")
+	// The above API call returns output containing list of versions as
+	//LayerVersions and a pagination token as NextMarker
+
+	// Extracting the list of versions and assigning it to a new variable
+	versionList := response.LayerVersions
+	totalVersion := len(versionList) // Number of total versions
+
+	// sorting the list in ascending order
+	sort.Slice(versionList, func(i, j int) bool {
+		return *versionList[i].Version < *versionList[j].Version
+	})
+
+	for i := 0; i < (totalVersion - 1); i++ {
```

Comment:
For this, we need to loop till second to last element, and not delete the last element (which is also the latest version of LayerVersion). For the same reason I haven't looped on complete list.

Reply:
Oh I see. Okay well I still think it's cleaner to do `for i := 0; i < len(versionList) - 1; i++ {`, rather than having a floating variable that's not used elsewhere. But it's a nit, so it's up to you

Reply:
Done